src/librustdoc/html/render/mod.rs

```diff
@@ -108,8 +108,8 @@ crate struct IndexItem {
 /// A type used for the search index.
 #[derive(Debug)]
 crate struct RenderType {
-    name: Option<String>,
-    generics: Option<Vec<TypeWithKind>>,
+    name: String,
+    generics: Vec<TypeWithKind>,
```

Comment:
Can this and similar refactorings have any effect on behavior? I found it quite bizarre that the search-index representation of a type can exist without a name.

Reply:
It's possible in case you have something like `T: Display + Whatever`. In this case, since `T` isn't a type, we don't keep its name. Please revert this change.

Reply:
This change didn't cause any tests to fail though. Could you please write a test case that fails with this refactoring? Otherwise it'll be hard to not break this code accidentally.

Reply:
Please add one then. Something like (in `rustdoc-js`):

```rust
pub fn foo<Something: Debug>(s: Something) {}
```

Then look for `Something`. You shouldn't have any result.